File: programmers/쇠막대기.cpp
#include <string>
#include <queue>
#include <vector>

using namespace std;

int solution(string arrangement) {
    int answer = 0;
    int flag=0;
    queue<char> q;
    for(int i=0;i<arrangement.size();i++){
     if(arrangement[i]=='('){
        q.push('(');
     }
        else{
            q.pop();
            if(arrangement[i-1]=='(')
                answer+= q.size();
            else
                answer+=1;
        }
            
    }
    return answer;
}

File: programmers/쇠막대기.java
class Solution {
    public int solution(String arrangement) {
        int answer = 0;
        int flag = 0;
        for (int i = 0; i < arrangement.length(); i++) {
            if (arrangement.charAt(i) == '(') {
                flag++;
            } else {
                flag--;
                if (arrangement.charAt(i - 1) == '(')
                    answer += flag;
                else
                    answer += 1;
            }
        }
        return answer;
    }
}